### PR TITLE
enable --offline mode for email again

### PR DESCRIPTION
**Pull request by kalikaneko**

### CODE REVIEW

**Comment by kalikaneko** [comment]
this works for enabling the imap service, which is the only one that currently *kinda* supports offline mode.
(but would need some more work to re-enable the changes in the status icons: login widget should say "offline mode", and mail should be "on")
We can leave that for a later pullreq.

File: src/leap/bitmask/backend/backend_proxy.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import zmq
 
 from leap.bitmask.backend.api import API, STOP_REQUEST, PING_REQUEST
+from leap.bitmask.backend.settings import Settings
```
Comment (ivanalejandro0) [inline src/leap/bitmask/backend/backend_proxy.py]: It looks that you're only creating the Settings object but not using anywhere, am I missing something?

File: src/leap/bitmask/backend/settings.py [hunk 2]
```diff
@@ -122,18 +122,17 @@ def set_selected_gateway(self, provider, gateway):
         self._settings.set(provider, self.GATEWAY_KEY, gateway)
         self._save()
 
-    def get_uuid(self, username):
+    def get_uuid(self, full_user_id):
         """
         Gets the uuid for a given username.
 
         :param username: the full user identifier in the form user@provider
```
Comment (ivanalejandro0) [inline src/leap/bitmask/backend/settings.py]: fix docstrings to reflect parameter name change.

File: src/leap/bitmask/backend/settings.py [hunk 3]
```diff
@@ -122,18 +122,17 @@ def set_selected_gateway(self, provider, gateway):
...
         :type username: basestring
         """
-        leap_assert("@" in username,
+        leap_assert("@" in full_user_id,
                     "Expected username in the form user@provider")
-        user, provider = username.split('@')
-
-        return self._get_value(provider, username, "")
+        username, provider = full_user_id.split('@')
+        return self._get_value(str(provider), str(full_user_id), "")
```
Comment (ivanalejandro0) [inline src/leap/bitmask/backend/settings.py]: is the typecast needed? I think that the ConfigParser object returns str objects.

File: src/leap/bitmask/backend/backend_proxy.py (hunk 1 above)
Comment (kalikaneko) [inline src/leap/bitmask/backend/backend_proxy.py]:
```
/home/kali/leap/bitmask_client/src/leap/bitmask/gui/mainwindow.py
[1] 129:        self._backend_settings = self._backend.settings
[2] 1320:            uuid = self._backend_settings.get_uuid(full_user_id)
```

File: src/leap/bitmask/backend/settings.py (hunk 2 above)
Comment (kalikaneko) [inline src/leap/bitmask/backend/settings.py]: fixed

File: src/leap/bitmask/backend/settings.py (hunk 3 above)
Comment (kalikaneko) [inline src/leap/bitmask/backend/settings.py]: no it isn't, I must have done that while debugging...
fixed!

**Comment by ivanalejandro0** [comment]
Is this for/related https://leap.se/code/issues/5909 ? ... should we add a changelog?

**Comment by kalikaneko** [comment]
> Is this for/related https://leap.se/code/issues/5909 
no, not really. that should be worked on its own. this is only the old login bypass, to avoid doing the remote login + soledad sync if the only thing we want is to access the local mail store.
intended mainly for debug purposes, so I'm inclined not to announce it in the changelog as an user-facing feature.

**Comment by ivanalejandro0** [comment]
merged on `develop`.

**Pull request closed without merging**